File: src-tauri/src/models/stored_event.rs
use serde::{Deserialize, Serialize};
use sqlx::FromRow;

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
#[serde(rename_all = "camelCase")]
pub struct StoredEvent {
    pub sequence_id: i64,
    pub stream_id: String,
    pub branch_id: String,
    pub kind: String,
    pub payload: String,
    pub version: i64,
    pub created_at: i64,
}

File: src/domain/schema/stored-event.ts
import { Schema } from 'effect'

export const StoredEvent = Schema.Struct({
  sequenceId: Schema.Number,
  streamId: Schema.String,
  branchId: Schema.String,
  kind: Schema.String,
  payload: Schema.String,
  version: Schema.Number,
  createdAt: Schema.Number,
})
export type StoredEvent = typeof StoredEvent.Type
